kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -166,14 +235,62 @@ private static DataType convertToPhysicalType(
         return logicalType;
     }
 
-    /** Returns the physical name for a given {@link StructField} */
-    public static String getPhysicalName(StructField field) {
-        if (field.getMetadata().contains(COLUMN_MAPPING_PHYSICAL_NAME_KEY)) {
-            return (String) field
-                .getMetadata()
-                .get(COLUMN_MAPPING_PHYSICAL_NAME_KEY);
-        } else {
-            return field.getName();
+    private static boolean allowMappingModeChange(String oldMode, String newMode) {
+        // only upgrade from none to name mapping is allowed
+        return oldMode.equals(newMode) ||
+                (COLUMN_MAPPING_MODE_NONE.equals(oldMode) &&
```

Comment:
would that enum be in `ColumnMapping` or in `TableConfig`? No worries, I can add that enum. However, that feels like it would be better to do in a follow-up PR in order to not increase the diff here, wdyt?

Reply:
enum in either places is fine.